tools/config-generator/periodic_config_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"gopkg.in/yaml.v2"
+)
+
+func TestClone(t *testing.T) {
+	SetupForTesting()
+	base := baseProwJobTemplateData{OrgName: "org-name"}
+	data := periodicJobTemplateData{
+		Base:            base,
+		PeriodicJobName: "periodic-job-name",
+		CronString:      "cron-string",
+		PeriodicCommand: []string{"string-a", "string-b"},
+	}
+	if diff := cmp.Diff(data.Clone(), data); diff != "" {
+		t.Errorf("Incorrect output for empty string: (-got +want)\n%s", diff)
+	}
+}
+
+func TestGetUTCtime(t *testing.T) {
+	SetupForTesting()
+	for i := 0; i < 24; i++ {
+		utcTime := getUTCtime(i)
+		expected := (i + 7) % 24
+		if utcTime != expected {
+			t.Errorf("Expected %d, got %d", expected, utcTime)
+		}
+	}
+}
+
+func TestCalculateMinuteOffset(t *testing.T) {
+	SetupForTesting()
+	out1 := calculateMinuteOffset("foo")
+	out2 := calculateMinuteOffset("foo")
+	if out1 != out2 {
```

Comment:
```suggestion
	if diff := cmp.Diff(out1, out2); diff != "" {
```

Would this get better error message?

Reply:
Done.